node/src/reactor/participating.rs

```diff
@@ -586,10 +586,10 @@ impl reactor::Reactor for Reactor {
             match chainspec.protocol_config.activation_point {
                 ActivationPoint::Genesis(genesis_timestamp) => {
                     // Check that no blocks exist in storage so we don't try to run genesis again on
-                    // an existing blockchain
-                    if let Some(first_block_header) = storage.read_block_header_by_height(1)? {
+                    // an existing blockchain.
+                    if let Some(highest_block_header) = storage.read_highest_block_header()? {
```

Comment:
Isn't that equivalent though? If we have `n-th` block or `1st` we would be overwriting a chain.

Reply:
If it's equivalent then this whole PR is unnecessary.

Reply:
The node may not necessarily have the first block but it could still have a highest block.  For instance if someone uses fast sync with the trusted hash of the 100th switchblock, then he or she will never download the 0th or 1st block-header (according to the flow in `run_fast_sync_task`).

Reply:
That makes sense, you're right. I didn't think of it.